demo_nodes_cpp/src/services/add_two_ints_server.cpp

```diff
@@ -16,29 +16,49 @@
 #include <cstdio>
 #include <memory>
 #include <string>
+#include <vector>
 
 #include "rclcpp/rclcpp.hpp"
+#include "rclcpp_components/register_node_macro.hpp"
 #include "rcutils/cmdline_parser.h"
 
 #include "example_interfaces/srv/add_two_ints.hpp"
 
-void print_usage()
+namespace demo_nodes_cpp
 {
-  printf("Usage for add_two_ints_server app:\n");
-  printf("add_two_ints_server [-s service_name] [-h]\n");
-  printf("options:\n");
-  printf("-h : Print this help function.\n");
-  printf("-s service_name : Specify the service name for this server. Defaults to add_two_ints.\n");
-}
-
 
 class ServerNode : public rclcpp::Node
 {
 public:
-  explicit ServerNode(const std::string & service_name)
-  : Node("add_two_ints_server")
+  explicit ServerNode(const rclcpp::NodeOptions & options)
+  : Node("add_two_ints_server", options)
   {
+    setvbuf(stdout, NULL, _IONBF, BUFSIZ);
+    std::vector<std::string> args = options.arguments();
+    if (find_command_option(args, "-h")) {
+      print_usage();
+      rclcpp::shutdown();
+    } else {
+      std::string tmptopic = get_command_option(args, "-s");
```

Comment:
Same comment as above. All of the CLI argument parsing logic can be removed now or in a follow-up if we want to backport this PR.